src/main.cc

```diff
@@ -939,6 +940,11 @@ main(int argc, char **argv)
 		if (start_loop) {
 			say_info("entering the event loop");
 			systemd_snotify("READY=1");
+			if (is_console_exited) {
+				printf("Exited console. Type Ctrl-C to exit "
+				       "Tarantool.\n");
```

Comment:
If it's okay, we should probably also prohibit calling `console.start()` *after* entering the event loop because it wouldn't work properly anyway.

Reply:
I like that `console.start()` is just doing its job and return to where it was called. If it was a top level call (`tarantool -i` case), it means exitting. OK. If it is called from a script, it returns to the script. OK too.

The session type can be saved before going to a REPL and returned back after exitting from the REPL.

The root of the confusing behavior is not about console at all. Tarantool can run a script for two distinct purposes: just execute a script or start a service (usually `box.cfg()`). It should exit after the script in the first case and shouldn't in the second case.

The criteria how tarantool differentiate the cases is quite obscure (`ev_activecnt()` after executing the script gives a number above it did before the script). AFAIU, it counts watchers.

It works surprisingly good, though. So, maybe we should just let user know what actually prevents tarantool from going down? Say, print a list of fibers with their names?

We can make the criteria much simpler: say, just add an option, whenther to wait for background jobs (and don't wait by default). However, I think that it would be too breaking change even for 3.0. There are a lot applications that expect that tarantool will not exit after the script end. In fact, all the applications leans on it.

We just introduced the new application startup flow (our declarative config effort) and I think that we're not in the point, where we can declare that scripts are just scripts. Maybe it'll be OK for 4.0.

Anyway, the topic is complicated and I like to discuss it voicely.

Reply:
We had a f2f discussion with @Gumix and @locker. The key point is that `tarantool -i script` works good[^1], so it looks like there are no reasons to use `require('console').start()` within `-e` option argument or a script code.

Personally I think that a warning is enough. Ideally, with a list of fibers, but it is better than nothing even without the list.

[^1]: `tarantool script -i` ignores `-i`. It looks like a bug.